[FIX][AERIE-1829] Satisfying activities missing

adrienmaillard opened the pull request:
* **Tickets addressed:** AERIE-1829
* **Review:** By commit  <!-- Choose from: "by commit", "by file" -->
* **Merge strategy:** Merge (no squash)  <!-- Choose from: "merge (no squash)", "squash and merge" -->

## Description
<!-- What approach was taken to satisfy the ticket being addressed? What should reviewers be aware of? -->
Activities that cannot be associated to their creator goal (which happens if you run the scheduler several times in the same plan as there is no tracking what activity were created by which goal in previous runs) were considered to have Solely custody. 

If you ran the scheduler twice without activity created, you would not see the satisfying activities associated to the goal. 

We now allow goal-activity association even if the activity does not have an explicit creator goal defined.   

## Verification
<!-- How were the changes validated? Were any automated tests added, updated, removed, or re-baselined? -->
With a test specifically designed to reproduce the issue in `SimulationFacadeTest`.

## Documentation
<!-- What documentation was invalidated by these changes? Which artifacts should reviewers check for accuracy and completeness? -->
None.

## Future work
<!-- What next steps can we anticipate from here, if any? -->
If goal/activity custody is going to be a feature, there will be a need to track from run to run what goals created which activity. And think about use cases where for example a user creates a goal, schedules activities and then deletes the goal. What happens to these activities and their custody?

adrienmaillard committed:
Fix issue with association

Activities that cannot be associated to their creator goal (which happens if you run the scheduler several times in the same plan as there is no tracking what activity were created by which goal in previous runs) were considered to have Solely custody.

camargo commented:
When trying to run this recurrence goal twice on an empty plan:

```ts
export default (): Goal => Goal.ActivityRecurrenceGoal({
  activityTemplate: ActivityTemplates.BakeBananaBread({ temperature: 325.0, tbSugar: 2, glutenFree: false }),
  interval: 12 * 60 * 60 * 1000 * 1000 // 1/2 day in microseconds
})
```

I'm now seeing this error in the scheduler:

```json
{"status":"failed","reason":"Cannot compute end time: activity instance does not have a duration yet"}
```

Can anyone reproduce on their end?

adrienmaillard committed:
Pull activity durations after simulation of initial plan

Activities of the initial plan are simulated but their durations were not pulled back to activity instance objects.

adrienmaillard commented:
> When trying to run this recurrence goal twice on an empty plan:
> 
> ```ts
> export default (): Goal => Goal.ActivityRecurrenceGoal({
>   activityTemplate: ActivityTemplates.BakeBananaBread({ temperature: 325.0, tbSugar: 2, glutenFree: false }),
>   interval: 12 * 60 * 60 * 1000 * 1000 // 1/2 day in microseconds
> })
> ```
> 
> I'm now seeing this error in the scheduler:
> 
> ```json
> {"status":"failed","reason":"Cannot compute end time: activity instance does not have a duration yet"}
> ```
> 
> Can anyone reproduce on their end?

Activities of the initial plan were simulated but their durations were not pulled back from the simulation which caused this error. It has been fixed by the second commit of this PR.

adrienmaillard committed:
Fix issue with detection of zero-duration activities

camargo reviewed (approved):
LGTM. Tested and works on front-end.
